pfio/v2/s3.py

```diff
@@ -398,12 +408,14 @@ def stat(self, path):
             return S3ObjectStat(key, res)
         except ClientError as e:
             if e.response['Error']['Code'] == '404':
+                if self.isdir(path):
+                    return S3PrefixStat(key)
```

Comment:
I don't think this is a good design - in cases like kind of a "non-existent" directory (somewhat hard to define), objects in a bucket like
- `foo/bar.txt` and
- `somefile.txt` 

In this case, `s3.isdir('foo')` would be true and also false for `s3.isdir('somefile.txt')`, which is nice, but this code will also return true for `s3.isdir('whatever-nothing')`, which is just nothing.

Reply:
> but this code will also return true for s3.isdir('whatever-nothing'), which is just nothing.

I think (and I confirmed locally) that `s3.isdir('whatever-nothing')` returns false because this PR handles "common-prefix" as directories. 

on the other hand, I think it's not the best choice in terms of performance. It always calls list-object API when the server returns 404 status code.

Reply:
Sorry, my bad! I misread the code.